### PR TITLE
[RELAY] Pass infra cleanup

### PR DESCRIPTION
**Pull request by tqchen**
Followup of https://github.com/dmlc/tvm/pull/3319

- Fix the parameter order of PassInfo
- Expose the basic primitive functions when creating passes.

### CODE REVIEW

**Comment by tqchen** [comment]
cc @zhiics

File: python/tvm/relay/transform.py [hunk 1]
```diff
@@ -39,19 +35,19 @@ class PassInfo(RelayNode):
...
+        The pass name.
+
     required : List[str]
         The list of passes that are required by a certain pass.
     """
 
-    def __init__(self, name, opt_level, required=None):
-        self.__init_handle_by_constructor__(_transform.PassInfo, name, opt_level,
-                                            required)
+    def __init__(self, opt_level, name, required=None):
```
Comment (jroesch) [inline python/tvm/relay/transform.py]: Why should we put opt_level first? I don't even think most passes should be on a standard opt-level, we end up in the current system we have where people arbitrarily assign pass numbers to passes and when passes are non-robust we just bump them to a higher pass number.

Comment (jroesch) [reply python/tvm/relay/transform.py]: We should focus on names and enabling passes by name in most cases imo.

Comment (tqchen) [reply python/tvm/relay/transform.py]: I made the change just to be consistent with the rest of the positional argument, we can debate whether it is a good idea to put name or opt_level first. I am fine either way

**Pull request merged**